dist/tools/whitespacecheck/check.sh

```diff
@@ -23,28 +25,46 @@ else
     fi
 fi
 
+# sets
+# - LOG to tee output into for later parsing
+# - LOGFILE to parse GitHub annotations into
+github_annotate_setup
 
 # select files to check
 if [ -z "${BRANCH}" ]; then
     BRANCH=$(git rev-list HEAD | tail -n 1)
 fi
 
 git -c core.whitespace="tab-in-indent,tabwidth=4" \
-    diff --check "$(git merge-base "${BRANCH}" HEAD)" -- *.[ch] ${IGNORE}
-
+	diff --check "$(git merge-base "${BRANCH}" HEAD)" -- *.[ch] ${IGNORE} \
+            | ${LOG}
 RESULT=$?
 
 # Git regards any trailing white space except `\n` as an error so `\r` is
 # checked here, too
 git -c core.whitespace="trailing-space" \
-    diff --check "$(git merge-base "${BRANCH}" HEAD)" -- . ${IGNORE}
+    diff --check "$(git merge-base "${BRANCH}" HEAD)" -- . ${IGNORE} \
+   	         | ${LOG}
 
 TRAILING_RESULT=$?
 
+if github_annotate_is_on; then
+    grep "^.\+:[0-9]\+:" "${LOGFILE}" | while read line; do
+        FILENAME=$(echo "${line}" | cut -d: -f1)
+        LINENR=$(echo "${line}" | cut -d: -f2)
+        DETAILS=$(echo "${line}" | cut -d: -f3- |
+                  sed -e 's/^[ \t]*//' -e 's/[ \t]*$//')
+        github_annotate_error "${FILENAME}" "${LINENR}" "${DETAILS}"
```

Comment:
```suggestion
        LINENUM=$(echo "${line}" | cut -d: -f2)
        DETAILS=$(echo "${line}" | cut -d: -f3- |
                  sed -e 's/^[ \t]*//' -e 's/[ \t]*$//')
        github_annotate_error "${FILENAME}" "${LINENUM}" "${DETAILS}"
```
The parameter is called like this in github_annotate_error/warning functions.

Reply:
Done.